### DISCUSSION TITLE
How can I add Textfield to the alert while using `AlertState`

### DISCUSSION THREAD

**Discussion by x-0o0** [pointfreeco/swift-composable-architecture#2164] [Q&A] [answered]
I'm trying to add TextField to the alert view.
In the vanilla SwiftUI, it could be resolved like below:
```swift
.alert("Enter your name", isPresented: $showingAlert) {
    TextField("Enter your name", text: $name)
    Button("OK", action: submit)
}
```
However, while using `AlertState` in `prerelease/1.0`, I'm still struggling with how to customize the alert.
*Could you give me some hint for this?*

Here is some part of my code:
```swift
// RoomListFeature.swift
var body: some ReducerOf<Self> {
    Reduce<State, Action> { state, action in
        switch action {
        case .startRoomButtonTapped:  // when the user tapped button to start a NEW room
            state.alert = AlertState {
                TextState("Start a new room?") // 🤔 I want to add text field to receive the name of the new room.
            } actions: {
                ButtonState(action: .startRoom) {
                    TextState("Start")
                }
            }
            return .none
        case .alert(.presented(.startRoom)): // when the user tapped "Start" button on the alert
            state.room = RoomReducer.State(room: Room(...))
            return .none

        case .alert:
            return .none
        }
    }
    .ifLet(\.$alert, action: /Action.alert)
}
```

```swift
// RoomList.swift
struct RoomList: View {
    let store: StoreOf<RoomListFeature>

    var body: some View {
        WithViewStore(self.store, observe: { $0 }) { viewStore in
            List { 
                // ... 
            }
            .alert(
                store: self.store.scope(
                    state: \.$alert,
                    action: { .alert($0) }
                )
            ) // 🤔 No additional parameter to customize the alert view
        }
    }
}
```

**Comment by x-0o0** [comment]
This is duplicated with discussion #1357 
Is there any updates on it?

**Reply by mbrandonw** [reply]
Hi @jaesung-0o0, there is no update on this currently. In order to use text fields in alerts you will need to use the vanilla SwiftUI tools.

**Reply by stephencelis** [reply, marked as answer]
I think @kamcma's solution of using Swift's `alert` API directly is still the way to go for now:

https://github.com/pointfreeco/swift-composable-architecture/discussions/2048#discussioncomment-5724309

We'd love to ship a `TextFieldState` in the future, but at the end of the day it'll be a race to maintain `AlertState` compatibility with all the alert features Apple provides in the future, so dropping down to the core APIs will always unblock you.

**Reply by x-0o0** [reply]
@mbrandonw @stephencelis Thanks to both of you for the answers
